File: 2024/May/4-2 2441 Largest Positive Integer That Exists With Its Negative/Set Approach/LargestMatch.py
from typing import List

class Solution:
    def findMaxK(self, nums: List[int]) -> int:
        positives = set()
        negatives = set()
        num = -1

        for i in nums:
            if i > 0:
                if i in negatives:
                    if i > num:
                        num = i
                else:
                    positives.add(i)
            else :
                pos = abs(i)
                if pos in positives:
                    if pos > num:
                        num = pos
                else:
                    negatives.add(pos)
        
        return num
                

File: 2024/May/4-2 2441 Largest Positive Integer That Exists With Its Negative/Set Approach/LargestMatch.java
import java.util.HashSet;

class Solution {
    public int findMaxK(int[] nums) {
        HashSet<Integer> positives = new HashSet<Integer>();
        HashSet<Integer> negatives = new HashSet<Integer>();
        int num = -1;
        
        for(int i : nums) {
            if(i > 0) {
                if(negatives.contains(i)) {
                    if(i > num) {
                        num = i;
                    }
                }
                else {
                    positives.add(i);
                }
            }
            else {
                int pos = Math.abs(i);
                if(positives.contains(pos)) {
                    if(pos > num) {
                        num = pos;
                    }
                }
                else {
                    negatives.add(pos);
                }
            }
        }

        return num;
    }
}